spring-ai-openai/src/main/java/org/springframework/ai/openai/api/OpenAiApi.java

```diff
@@ -74,8 +75,8 @@ public OpenAiApi(String openAiToken) {
 	 * @param restClientBuilder RestClient builder.
 	 */
 	public OpenAiApi(String baseUrl, String openAiToken, RestClient.Builder restClientBuilder) {
-
-		this.objectMapper = new ObjectMapper();
+		// Use the same ObjectMapper for WebClient's response as the one used in RestClient.
+		this.objectMapper = Jackson2ObjectMapperBuilder.json().build();
```

Comment:
This is interesting, need to dig a bit more. 
The recent streaming response parsing failure was due to the unexpected (for me) appearance of the `LogProbs` in the chunk response.  Not sure if i missed it or OpenAI introduced it silently. Hope it is the former. 
I deliberately didn't configure the objectmapper to ignore unknown fields  so i can catch changes. But didn't expect they will come without announcement   and also didn't consider that the webclient and restclient converters are likely ignoring such fields.

Reply:
I encountered an error during code review. It appears that LogProbs were added at some point. After checking OpenAI's API specification at https://platform.openai.com/docs/api-reference/chat/object, I noticed the presence of LogProbs. However, in the response for the ChatCompletion - Choice record when the stream is false, LogProbs are not included. To address this issue, I realized that a similar approach is needed for WebClient, similar to the ObjectMapper in RestClient, to ignore unknown fields. Therefore, I added the same configuration in the commit.

Instead of adding LogProbs to ChatCompletion - Choice and ChatCompletionChunk record, considering the continuous changes and potential additions or deprecations in OpenAI's spec, I believe it's more beneficial for developers using Spring AI to configure the ObjectMapper to ignore unknown fields. This way, they won't have to deal with too many specific options. Hence, I implemented the ObjectMapper configuration in the current commit.

Reply:
We already added the LogProbs in another commit. 
But with this PR I'll configure the object mapper to tolerate unknown properties. 
Would live out the Jackson2ObjectMapperBuilder for now.